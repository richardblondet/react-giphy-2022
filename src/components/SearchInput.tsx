import * as React from 'react';
import {
  Divider, IconButton, InputBase, InputProps, Paper, Tooltip,
} from '@mui/material';
import { FiSearch, FiTrendingUp } from 'react-icons/fi';

interface SearchInputProps extends InputProps {
  loadTrending: () => void;
  onSearchClick: () => void;
}

export default function SearchInput({
  loadTrending,
  onSearchClick,
  ...inputProps
}: SearchInputProps) {
  return (
    <Paper
      component="div"
      sx={{
        p: '2px 4px', display: 'flex', alignItems: 'center', width: '500px', marginBottom: '3em', mx: 'auto',
      }}
    >
      <InputBase
        sx={{ ml: 1, flex: 1 }}
        placeholder="Search Giphy Library"
        inputProps={{ 'aria-label': 'search giphy library' }}
        {...inputProps}
      />
      <Tooltip title="Search">
        <IconButton type="submit" sx={{ p: '10px' }} aria-label="search" onClick={onSearchClick}>
          <FiSearch />
        </IconButton>
      </Tooltip>
      <Divider sx={{ height: 28, m: 0.5 }} orientation="vertical" />
      <Tooltip title="View trending">
        <IconButton color="primary" sx={{ p: '10px' }} aria-label="directions" onClick={loadTrending}>
          <FiTrendingUp />
        </IconButton>
      </Tooltip>
    </Paper>
  );
}
