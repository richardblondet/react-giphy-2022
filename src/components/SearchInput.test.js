
import React from 'react';
import renderer from 'react-test-renderer';
import SearchInput from './SearchInput';

test('Link changes the class when hovered', () => {
  const component = renderer.create(
    <SearchInput
      id="search-input"
      value={"cats"}
      onChange={() => void 0}
      loadTrending={() => void 0}
      onSearchClick={() => void 0}
    />,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});