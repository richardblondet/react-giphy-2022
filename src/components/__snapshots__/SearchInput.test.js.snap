// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link changes the class when hovered 1`] = `
<form
  className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1vbswbz-MuiPaper-root"
>
  <div
    className="MuiInputBase-root MuiInputBase-colorPrimary css-piqts5-MuiInputBase-root"
    onClick={[Function]}
  >
    <input
      aria-label="search giphy library"
      className="MuiInputBase-input css-yz9k0d-MuiInputBase-input"
      id="search-input"
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="Search Giphy Library"
      type="text"
      value="cats"
    />
  </div>
  <button
    aria-label="search"
    aria-labelledby={null}
    className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-1gtpq4r-MuiButtonBase-root-MuiIconButton-root"
    data-mui-internal-clone-element={true}
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="submit"
  >
    <svg
      fill="none"
      height="1em"
      stroke="currentColor"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="2"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        cx="11"
        cy="11"
        r="8"
      />
      <line
        x1="21"
        x2="16.65"
        y1="21"
        y2="16.65"
      />
    </svg>
    <span
      className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
  <hr
    className="MuiDivider-root MuiDivider-fullWidth MuiDivider-vertical css-jbezd2-MuiDivider-root"
  />
  <button
    aria-label="directions"
    aria-labelledby={null}
    className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary MuiIconButton-sizeMedium css-1bxg8jb-MuiButtonBase-root-MuiIconButton-root"
    data-mui-internal-clone-element={true}
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <svg
      fill="none"
      height="1em"
      stroke="currentColor"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="2"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <polyline
        points="23 6 13.5 15.5 8.5 10.5 1 18"
      />
      <polyline
        points="17 6 23 6 23 12"
      />
    </svg>
    <span
      className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</form>
`;
